### DISCUSSION TITLE
How can I get the total energy consumed by the house

### DISCUSSION THREAD

**Discussion by vinchz31** [addiejanssen/domoticz-solaredge-modbustcp-plugin#53] [Q&A] [answered]
Hello,
First of all, thank you very much for this module which works perfectly and on the first try!
I have a question (I could not find the answer in the discussions): how can we recover the total energy consumed by the house in addition to the energy produced by the inverter (I did not find this information in the different values returned) ? I would need this information (available on the API) to be able to trigger my elements when the consumption is lower than the production.
Thank you in advance for your feedback and help and thanks again for this wonderfull pluggin !
Best regards.
Vincent

**Comment by vinchz31** [comment]
Anyone can Help me ? 

I think there's a solution here : https://github.com/addiejanssen/domoticz-solaredge-modbustcp-plugin/discussions/13
...but it the plugin doesn't work for me...

Many thanks !
Vincent

**Comment by addiejanssen** [comment, marked as answer]
Hi Vincent,

The plugin will only provide the information shared by the inverter and attached meters (and batteries once that is working).
There is no functionality in the plugin to do these types of calculations.
Since every electrical installation can be different, it is impossible to implement a generic/re-usable solution.

You use the Domoticz scripting functionality to do the calculation that you are looking for.
I would recommend to look at https://www.domoticz.com/wiki/DzVents:_next_generation_Lua_scripting ; this allows you to collect the information from an inverter and meter and calculate the consumption applicable for your electrical installation.
If you need help with this, I would recommend [the Domoticz forum](https://www.domoticz.com/forum/).

**Comment by vinchz31** [comment]
Many Thanks for your answer addiejanssen and for you plugin !
I'm not a developer so I'm afraid I won't be able to handle this topic myself in ULA.
I'm talking with pfolcato in the topic https://github.com/addiejanssen/domoticz-solaredge-modbustcp-plugin/discussions/13 to try to use his update that brings up these values hoping he can find the origin of my problem.
Have a nice day !
